### PR TITLE
Rewrite Golemagg and fix 'Golemagg's Trust'

### PR DESCRIPTION
**Pull request by schell244**
## 🍰 Pullrequest
Some fixes I've made for the [RaspyWoW](https://www.raspywow.com/) project.
Update: removed unverified changes.

Implemented missing handling for spells:
- ~~Moss Covered Feet~~
- Golemagg's Trust
- ~~Bethor's Potion~~

~~Fixed implementation of spell:~~
- ~~Mark of Frost~~

Script of Golemagg is fully rewritten, with Golemagg's Trust now being implemented and also added the missing leash handling for his adds.

30 yards buff reach: [wowhead](https://www.wowhead.com/classic/spell=20553/golemaggs-trust)
100 yards leash distance: [Azerothcore](https://github.com/azerothcore/azerothcore-wotlk/blob/106cbe7fdf7d3b7b4906d04cafc40e61ff62d002/src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_golemagg.cpp#L182)

### Proof
- None

### How2Test
- None

### Todo / Checklist
- [ ] spell timers?
- [ ] aura 18943 correct?

### CODE REVIEW

File: sql/migrations/20241117202208_world.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+BEGIN
+DECLARE v INT DEFAULT 1;
+SET v = (SELECT COUNT(*) FROM `migrations` WHERE `id`='20241117202208');
+IF v = 0 THEN
+INSERT INTO `migrations` VALUES ('20241117202208');
+-- Add your query below.
+
+-- Moss Covered Feet - add SPELL_ATTR_EX_AURA_UNIQUE
+INSERT INTO `spell_mod` (`Id`, `procChance`, `procFlags`, `procCharges`, `DurationIndex`, `Category`, `CastingTimeIndex`, `StackAmount`, `SpellIconID`, `activeIconID`, `manaCost`, `Attributes`, `AttributesEx`, `AttributesEx2`, `AttributesEx3`, `AttributesEx4`, `Custom`, `InterruptFlags`, `AuraInterruptFlags`, `ChannelInterruptFlags`, `Dispel`, `Stances`, `StancesNot`, `SpellVisual`, `ManaCostPercentage`, `StartRecoveryCategory`, `StartRecoveryTime`, `MaxAffectedTargets`, `MaxTargetLevel`, `DmgClass`, `rangeIndex`, `RecoveryTime`, `CategoryRecoveryTime`, `SpellFamilyName`, `SpellFamilyFlags`, `Mechanic`, `EquippedItemClass`, `Comment`) VALUES
+(6870, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 67110928, -1, -1, -1, 1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 0, -1, -1, 'Moss Covered Feet - Add unique aura flag');
```
Comment (ratkosrb) [inline sql/migrations/20241117202208_world.sql]: Why is this necessary?

File: src/game/Spells/SpellAuras.cpp [hunk 2]
```diff
@@ -2062,8 +2072,18 @@ void Aura::HandleAuraDummy(bool apply, bool Real)
             case 23183:                                     // Mark of Frost
             {
                 if (m_removeMode == AURA_REMOVE_BY_DEATH)
-                    target->CastSpell(target, 23182, true, nullptr, this);
-                    return;
+                {
+                    if (Unit* caster = GetCaster())
+                    {
+                        // Azuregos
+                        // Only cast Mark of Frost on targets nearby when engaged
```
Comment (ratkosrb) [inline src/game/Spells/SpellAuras.cpp]: Is this blizzlike?

File: sql/migrations/20241117202208_world.sql (hunk 1 above)
Comment (schell244) [reply sql/migrations/20241117202208_world.sql]: Without this modification, the spell can be applied multiple times to a player. I have no proof that this is correct - I just remember it not being applied more than once. I can remove this line if that’s preferable.

File: src/game/Spells/SpellAuras.cpp (hunk 2 above)
Comment (schell244) [reply src/game/Spells/SpellAuras.cpp]: I also have no poof for this - needs to be confirmed.

File: sql/migrations/20241117202208_world.sql (hunk 1 above)
Comment (schell244) [reply sql/migrations/20241117202208_world.sql]: https://github.com/vmangos/core/issues/1223#issuecomment-1257117399 - maybe this can be achieved differently, but currently I don't have any other idea except adding the unique flag.
